src/coreclr/gc/gc.cpp

```diff
@@ -3218,6 +3218,35 @@ void gc_heap::fire_pevents()
     {
         time_info_32[i] = limit_time_to_uint32 (time_info[i]);
     }
+#ifdef DYNAMIC_HEAP_COUNT
+    if ((GCConfig::GetHeapCount() == 0) && (GCConfig::GetGCDynamicAdaptationMode() != 0))
+    {
+        uint32_t prev_sample_index = (dynamic_heap_count_data.sample_index + dynamic_heap_count_data_t::sample_size - 1) % dynamic_heap_count_data_t::sample_size;
+        assert (prev_sample_index < dynamic_heap_count_data_t::sample_size);
+        dynamic_heap_count_data_t::sample& sample = dynamic_heap_count_data.samples[prev_sample_index];
+        uint64_t gc_elapsed_time     = sample.gc_elapsed_time;
+        uint64_t soh_msl_wait_time   = sample.soh_msl_wait_time;
+        uint64_t uoh_msl_wait_time   = sample.uoh_msl_wait_time;
+        uint64_t elapsed_between_gcs = sample.elapsed_between_gcs;
+
+        // TODO, AndrewAu, is this the right time to fire these events?
+        //
+        // I get it, we need to reuse the existing GCGlobalHeapHistory_V4 event, so we had no choice
+        // but to fire it now, but now we have an option to do something different.
+        //
+        // GCEventFireDynamicHeapCountData is also computed once across all heaps - so this is not
+        // more global than that one
+        //
+        // Right now, this is fired more frequently than GCEventFireDynamicHeapCountData
+        // 
+        GCEventFireGlobalDynamicHeapCountData(
```

Comment:
Probably renaming the event name to include the fact that the metrics have information about the heuristics / metrics associated with computing the new heap count will make it more descriptive.